Notes.md

```diff
@@ -0,0 +1,10 @@
+1. Dereference : Grabbing the underlying value that is located in a particular address
+  * int x  = 10; 
+    int *p = &x;  // address of x
+
+    printf("%d\n", *p) // deference the pointer and grab 10 from x 
+                       // which p is pointing to. *p will print 10. 
+
+2. Heap - is where all your allocated memory variables live and variables can be passed from one function to the next. 
+
+3. Stack - This is where all your static variables live and so it does not live outside the scope of the function. 
```

Comment:
This isn't exactly right. Static variables aren't just those variables that live inside a single function's scope, though it does _include_ those. But static global variables in a file would also live inside of the stack, as well as statically-allocated arrays (those that are not initialized via a call to `malloc`).